students/gregdevore/lesson05/mailroom.py

```diff
@@ -0,0 +1,159 @@
+donors = {'Eleanor Shellstrop':[25.00, 57.00],
+            'Chidi Anagonye':[150.00, 300.00, 275.00],
+            'Tahani Al-Jamil':[2000.00,7500.00,12000.00],
+            'Jason Mendoza':[15.00,40.00,60.00],
+            'Mindy St. Claire':[500.00]}
+
+# Main program prompt
+prompt = '\n'.join(['','Welcome to The Good Place charity donor database.',
+            'Please select from the following options:',
+            '1 - Send a Thank You',
+            '2 - Create a Report',
+            '3 - Send letters to all donors',
+            '4 - Exit Program',
+            '','Input > '])
+
+def generate_email(donor_name, donation_amount, total_amount):
+    email_dict = {'donor_name':donor_name, 'donation_amount':donation_amount, 'total_amount':total_amount}
+    # Create formatted email that can be copied & pasted
+    email = ('\n'.join(['Dear {donor_name},','',
+    'Thank you for your generous donation of ${donation_amount:.2f}.',
+    'To date, you have donated a total of ${total_amount:.2f} to our charity.',
+    'Your contributions help new arrivals receive the highest quality care possible.',
+    'Please know that your donations make a world of difference!',
+    '','Sincerely,','The Good Place Team'])).format(**email_dict)
+    return(email)
+
+def write_thank_you(donor_name=None):
+    # Add donation for new or existing donor and compose 'Thank You' message.
+    # Donor name can be specified in case re-calling after invalid donation amount.
+    while True:
+        print()
+        # Prompt for name if not specified
+        if not donor_name:
+            name = input('Enter donor name (type \'list\' to see donors or \'quit\' to exit): ')
+        else:
+            name = donor_name
+        if name == 'list': # List donors
+            print('\nCurrent list of donors:\n')
+            for donor in donors:
+                print(donor)
+        elif name == 'quit': # Return to main prompt
+            return
+        else: # Get donations for donors
+            donations = donors.get(name,[])
+            # Prompt for donation amount
+            amount = input('Enter donation amount in dollars (type \'quit\' to exit): ')
+            # If the user wants to bail mid-entry, remove the donor that was just
+            # added (if they were new) and return to main prompt
+            if amount == 'quit':
+                return
+            else: # Otherwise, convert donation amount to float
+                # Capture error if 'amount' not convertable
+                try:
+                    amount = float(amount)
+                    if amount <= 0:
+                        print('Invalid amount, please enter a positive value.')
+                        # Re-call function with current name to re-prompt for amount
+                        write_thank_you(donor_name=name)
+                        # Need return statement here after return from recursive call
+                        return
+                except ValueError:
```

Comment:
it's best to catch ValueError around the line that can throw that error (line 59)